notfound/extension.py

```diff
@@ -194,6 +194,11 @@ def process_doc(self, app, doctree):
         if sphinx.version_info >= (3, 0, 0):
             metadata.update({'nosearch': True})
 
+    def merge_other(self, app, env, docnames, other):
+        """Merge in specified data regarding docnames from a different `BuildEnvironment`
+        object which coming from a subprocess in parallel builds."""
+        env.metadata.update(other.metadata)
```

Comment:
Do you have an example of other collector that works in parallel and implement this method? It would be good to put it here as reference as well.

Reply:
I could not find an example, so I read the documentation and printed out values while processing documents. Maybe @jakobandersen would be willing to look at it, because he diagnosed the problem here: https://github.com/sphinx-doc/sphinx/issues/8256

Reply:
I'm not familiar with what this extension is doing, so perhaps the following is irrelevant: technically you should not copy all data from ``other``, but only the data related to the documents in ``docnames``. However, I don't think I ever ran into a case where this in practice didn't mean "copy everything from ``other``", so maybe is really is a guarantee that ``other`` at most contains data about those documents.